src/index.ts

```diff
@@ -53,7 +53,7 @@ const stringify = (value: string = '') => {
 };
 
 const decode = (str: string): string => {
-  if (!str) return str;
+  if (!str || typeof str !== 'string') return str;
```

Comment:
Hi, thanks for your PR but I think we have a problem with `setCookie` function and only on backend side. 

For example, if we run this code on frontend side everything will be ok 
```
setCookie('MY_TEST_COOKIE', true);
console.log('MY TEST COOKIE HERE', getCookie('MY_TEST_COOKIE'));
```
and 
```
console.log(typeof getCookie('MY_TEST_COOKIE')) === string 
```

I think we need to figure out why a boolean value is set and not a string

Reply:
I figured it out a little and I think the problem is in line 109
It seems we need to use `stringify` function

```
const payload = { name: key, value: stringify(data), ...restOptions };
```
like in 132 line 
```
const cookieStr = serialize(key, stringify(data), { path: '/', ..._cookieOptions });
```

Reply:
Thank you for the review!🙌 I understand your perspective. In this PR (after my changes), setting cookies as a boolean or even another type like a number works correctly: the value is being set (server-side), and you can easily get it (server-side), but only with the same type as it was set with. Let's examine these examples: 


`    setCookie("MY_TEST_COOKIE", true, { req });
    console.log("MY TEST COOKIE HERE", typeof getCookie("MY_TEST_COOKIE", { req })); // boolean`
    
 `setCookie("MY_TEST_COOKIE", 10, { req });
    console.log("MY TEST COOKIE HERE", typeof getCookie("MY_TEST_COOKIE", { req })); // number`
    
  `setCookie("MY_TEST_COOKIE", 'test string', { req });
    console.log("MY TEST COOKIE HERE", typeof getCookie("MY_TEST_COOKIE", { req })); // string`
    
  Setters and getters use cookies from next/headers and the req object from Next.js. They also behave accordingly. I suggest we maintain this approach and update the rest of the codebase to ensure consistent type handling.

Therefore, my proposal is to merge this PR as a fix for setting values server-sode and modify the rest of the codebase to adhere to the native Next.js behavior.

Reply:
Sorry for delay. 

Cookie value is always string. 
Your example works as you say because you try to set and to get cookies during one request

For example. Let's imagine two API endpoints 
```ts
// api/set-cookie.ts 
export const GET = async (req: NextRequest) => {
      const res = new NextResponse();
      setCookie('test', true, { res, req });
      console.log(typeof getCookie('test', { res, req })); // ---> boolean
      return res;
};
```

```ts
// api/get-cookie.ts 
export const GET = async (req: NextRequest) => {
      const res = new NextResponse();
      console.log(typeof getCookie('test', { res, req })); // ---> string
      return res;
};
```

I think this behavior can lead unexpected results so better way is to always set string and always get string


Also, without "app router" this library transform value to string before setting. It will be better to keep the same behavior for different cases.

Reply:
Ok I got It! 🙌  I have done another request where all values should be set as string. Please check it out and test it.  #69

Reply:
So I think this PR should be closed